dpnp/backend/kernels/dpnp_krnl_sorting.cpp

```diff
@@ -87,33 +87,62 @@ template <typename _DataType>
 class dpnp_partition_c_kernel;
 
 template <typename _DataType>
-void dpnp_partition_c(const void* sort_array1_in, void* result1, const size_t kth, const size_t* shape, const size_t ndim)
+void dpnp_partition_c(void* array1_in, void* result1, const size_t kth, const size_t* shape_, const size_t ndim)
 {
-
-    cl::sycl::event event;
-
-    const _DataType* sort_arr = reinterpret_cast<const _DataType*>(sort_array1_in);
+    _DataType* arr = reinterpret_cast<_DataType*>(array1_in);
     _DataType* result = reinterpret_cast<_DataType*>(result1);
 
-    size_t size_ = 1;
-    for (size_t i = 0; i < ndim - 1; ++i)
+    if ((arr == nullptr) || (result == nullptr))
+    {
+        return;
+    }
+
+    size_t size = 1;
+    for (size_t i = 0; i < ndim; ++i)
     {
-        size_ *= shape[i];
+        size *= shape_[i];
     }
 
+    size_t size_ = size/shape_[ndim-1];
+
     if (size_ == 0)
     {
         return;
     }
 
+    auto arr_to_result_event = DPNP_QUEUE.memcpy(result, arr, size * sizeof(_DataType));
+    arr_to_result_event.wait();
+
+    for (size_t i = 0; i < size_; ++i)
+    {
+        size_t ind_begin = i * shape_[ndim-1];
+        size_t ind_end = (i + 1) * shape_[ndim-1] - 1;
+
+        _DataType matrix[shape_[ndim-1]];
+        for (size_t j = ind_begin; j < ind_end + 1; ++j)
+        {
+            size_t ind = j - ind_begin;
+            matrix[ind] = arr[j];
+        }
+        std::partial_sort(matrix, matrix + shape_[ndim-1], matrix + shape_[ndim-1]);
+        for (size_t j = ind_begin; j < ind_end + 1; ++j)
+        {
+            size_t ind = j - ind_begin;
+            arr[j] = matrix[ind];
+        }
+    }
```

Comment:
You can try parallelize this loop by adding `parallel_for` kernel in the future.

Also `std::partial_sort` is supported by `pstl` (see https://software.intel.com/content/www/us/en/develop/articles/get-started-with-parallel-stl.html) and `dpc++` + `pstl` as well (`dpl`, see: https://spec.oneapi.com/versions/0.5.0/oneAPI/Elements/onedpl/onedpl_root.html#).

This is just suggestions and it's better to do such optimization work on other PR(s).